src/Form/LinkBlockFormDataProvider.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\LinkList\Form;
+
+use PrestaShop\LinkList\Model\LinkBlock;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+
+class LinkBlockFormDataProvider implements FormDataProviderInterface
```

Comment:
yes I realized this, that's where I keep blocking for the moment I am not sure how to deal with these kind of CRUD form for now Do you have any example implementing this kind of form?
I had a look to your CQRS PR but it might be overkill in this case, and it is not ready yet anyway